### DISCUSSION TITLE
JavaScript code error

### DISCUSSION THREAD

**Discussion by aditikochhar** [community/community#21397] [Other Feature Feedback, Questions, & Ideas]
<p>var x = 1;<br>
function add Two(){<br>
var x = x +2;</p>
<p>}<br>
add Two();<br>
x = x+1;<br>
console.log(x);</p>
<p>I am getting an error when i run the code ? I think the answer is 1. Can someone please help ?</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hi <a class="mention" href="/aditikochhar">@aditikochhar</a> and welcome to the community!</p>
<p>There are several things going on here that is will cause errors:</p>
<ul>
<li>You cannot use spaces in function names. Try renaming your function to  <strong>addTwo</strong> instead of  <strong>add&lt;space&gt;Two</strong>.</li>
<li>You re-define  <strong>x</strong> inside your function by using the  <strong>var</strong> keyword. That way, the  <strong>x </strong> inside your function is a different  <strong>x</strong> than the  <strong>x</strong> in the first line.</li>
</ul>
<p>This should fix the errors mostly, however I also do recommend to pass the variable that needs the addition as a function argument as well as giving the variable of the product a different name (for clarity). Your code will look something like this then:</p>
<pre><code>var x = 1;
function addTwo(number) {
  return number + 2;

};
var y = addTwo(x);
y = y + 1;
console.log(y); // Should print the number 4 (1 + 2 + 1)
</code></pre>
<p>Again, these last changes are optional but do improve the quality of the code (at least in my opinion). Please let us know if you have any more questions.</p>
<p><strong>PS:</strong> next time, please choose a title that fits the topic and put your code in a code block.</p>
